### DISCUSSION TITLE
Missing types from `vitest/runners` module

### DISCUSSION THREAD

**Discussion by aleclarson** [vitest-dev/vitest#7077] [Ideas]
Any types used by `VitestRunner` or a subclass thereof should be exported from the `vitest/runners` module.

Some examples of currently missing types include:
- `SerializedConfig` (used by `VitestTestRunner` constructor)
- `Task`
- `TaskResult` (used by `Task.result` property)
- `ErrorWithDiff` (used by `TaskResult.errors` property)
- `VitestRunner` (extended by `VitestTestRunner` class)
- etc

This will make it more convenient to subclass an existing runner. Currently, I have to declare my own types, which is inconvenient and a bit noisy.

```ts
import { VitestTestRunner } from 'vitest/runners'

type SerializedConfig = ConstructorParameters<typeof VitestTestRunner>[0]
type Task = Parameters<VitestTestRunner['onAfterRunTask']>[0]
type ErrorWithDiff = Exclude<Exclude<Task['result'], void>['errors'], void>[number]

export default class extends VitestTestRunner {
  constructor(config: SerializedConfig) {
    super(config)

    /* ... */
  }

  onAfterRunTask(test: Task) {
    /* ... */
  }
}
```

**Comment by sheremet-va** [comment]
All these types are available from `vitest`

**Reply by aleclarson** [reply]
Ah great, thank you. This wasn't mentioned in the docs page about custom runners: https://vitest.dev/advanced/runner.html#test-runner

**Reply by aleclarson** [reply]
Also, I don't see `VitestRunner` exported anywhere?

~No `VitestExecutor` either.~ (Found this one in `vitest/execute`)

**Reply by sheremet-va** [reply]
If you are using a custom runner, I recommend importing types from `@vitest/runner` directly, or most of them are available from `vitest`. `vitest/runners` intentionally exports only runner implementations and not types, but I also added `VitestRunner` there in https://github.com/vitest-dev/vitest/pull/7240. `VitestExecutor` is also an internal class and should be used with caution, it doesn't follow semver
